public/app/features/alerting/unified/rule-list/filter/InlineSaveInput.tsx

```diff
@@ -0,0 +1,134 @@
+import { css } from '@emotion/css';
+import { useEffect, useRef } from 'react';
+import { useForm } from 'react-hook-form';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { t } from '@grafana/i18n';
+import { Box, IconButton, Input, Stack, Text, useStyles2 } from '@grafana/ui';
+
+import { SavedSearch, ValidationError, validateSearchName } from './SavedSearches.types';
+
+// ============================================================================
+// Inline Save Input (compact input with icon buttons)
+// ============================================================================
+
+export interface InlineSaveInputProps {
+  onSave: (name: string) => Promise<ValidationError | void>;
+  onCancel: () => void;
+  savedSearches: SavedSearch[];
+}
+
+interface FormValues {
+  name: string;
+}
+
+export function InlineSaveInput({ onSave, onCancel, savedSearches }: InlineSaveInputProps) {
+  const styles = useStyles2(getStyles);
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const {
+    register,
+    handleSubmit,
+    setError,
+    formState: { errors, isSubmitting },
+  } = useForm<FormValues>({
+    defaultValues: { name: '' },
+  });
+
+  // Focus input on mount
+  useEffect(() => {
+    inputRef.current?.focus();
+  }, []);
+
+  const onSubmit = async (data: FormValues) => {
+    const validationError = validateSearchName(data.name, savedSearches);
+    if (validationError) {
+      setError('name', { message: validationError });
+      return;
+    }
```

Comment:
I think we could use `react-hook-form` validation instead of doing it manually https://www.react-hook-form.com/api/useform/register